app/controllers/CampaignApi.scala

```diff
@@ -2,93 +2,90 @@ package controllers
 
 import com.gu.googleauth.AuthAction
 import model._
-import model.command._
-import model.reports._
 import play.api.Logger
-import play.api.libs.json.Json._
-import play.api.libs.json._
 import play.api.mvc._
 import repositories._
 import services.CampaignService
+import cats.syntax.either._
+import model.command._
+import play.api.libs.json.Json
 
 class CampaignApi(components: ControllerComponents, authAction: AuthAction[AnyContent])
   extends AbstractController(components) {
 
   def getCampaign(id: String) = authAction {
-    CampaignRepository.getCampaign(id) map { c =>
-      Ok(Json.toJson(c))
+    CampaignRepository.getCampaign(id) map { campaign =>
+      Ok(Json.toJson(campaign))
     } getOrElse NotFound
   }
 
   def getAllCampaigns() = authAction {
-    Ok(Json.toJson(CampaignRepository.getAllCampaigns()))
+    CampaignRepository.getAllCampaigns() match {
+      case Left(_)          => InternalServerError
```

Comment:
Is there anything here that could be caught in the error message?

Reply:
Yes there is. At the moment this was just done to get it to compile. @janua and I were talking about an approach earlier - this will come in a seperate PR.